client/app/components/date_range_picker.coffee

```diff
@@ -66,37 +64,26 @@ module.exports = DateRangePicker = React.createClass
         @props.onDateFilter start, end
 
 
-    reset: ->
-        @setState
-            isActive:  false
-            startDate: ''
-            endDate:   '',
-            @filterize
-
-
-    presetYesterday: ->
-        @setState
-            isActive:  true
-            startDate: moment().subtract(1, 'day').format(momentFormat)
-            endDate:   moment().subtract(1, 'day').format(momentFormat),
-            @filterize
+    onReset: ->
+        @changeDates()
 
+    onYesterday: ->
+        @changeDates 'day'
 
-    presetLastWeek: ->
-        @setState
-            isActive:  true
-            startDate: moment().subtract(1, 'week').format(momentFormat)
-            endDate:   moment().format(momentFormat),
-            @filterize
+    onLastWeek: ->
+        @changeDates 'week'
 
+    onLastMonth: ->
+        @changeDates 'month'
 
-    presetLastMonth: ->
-        @setState
-            isActive:  true
-            startDate: moment().subtract(1, 'month').format(momentFormat)
-            endDate:   moment().format(momentFormat),
-            @filterize
+    changeDates: (type) ->
+        value = moment().subtract(1, type).startOf(type) if type
```

Comment:
The choice is between : 
@poupotte && @kelukelu : I can resume the issue like : 
User click on 'last month' and all messages from last month only are displayed in the list
or
User click on 'last month' and all messages from last month to this day are displayed in the list